### PR TITLE
Show message_form_group_empty only if...

### PR DESCRIPTION
**Pull request by RobinDev**
Show message_form_group_empty only if there is nothing to show in the box.

I am targeting this branch, because everything backwards compatible.

## Changelog

```markdown
### Changed
- Show message_form_group_empty only if there is nothing to show in the box.
```

### CODE REVIEW

**Comment by RobinDev** [comment]
It permits to add a new box with a description and no fields without this message.

Used in the two last box here : 
screenshot

**Comment by VincentLanglet** [comment]
What if someone still want the description AND the empty message when there is no field ?

A better solution could be an option to override the empty message text (and the empty message translation domain).
And if you use the `false` value there is no message.

```
 ->with('General', array(
    'description' => 'This section contains general settings for the web page',
    'empty_message' => ...,
    'empty_message_translation_domain' => ...,
)
```

**Comment by RobinDev** [comment]
Nice suggestion avoiding bc break. I push a commit.

**Comment by VincentLanglet** [comment]
Nice ! 
Now, since you allow to override the `empty_message`, I think we should allow to override the translation domain.

And since `null == false` in twig, I recommend to use same as for the check.

```
{% if form_group.empty_message is not defined or form_group.empty_message not same as(false) %}
    <em>{{ form_group.empty_message|default('message_form_group_empty')|trans({}, form_group.empty_message_translation_domain|default('SonataAdminBundle')) }}</em>
{% endif %}
```

**Comment by VincentLanglet** [comment]
Plus you should not use `is defined`, and instead use default values.
See here: https://github.com/sonata-project/SonataAdminBundle/blob/a13f2735b7d3c5d5cb7c9b66fcef47df111dc02f/src/Mapper/BaseGroupedMapper.php#L75

File: src/Resources/views/CRUD/base_edit_form_macro.html.twig [hunk 1]
```diff
@@ -20,7 +20,9 @@
                         {% for field_name in form_group.fields|filter(field_name => form[field_name] is defined) %}
                             {{ form_row(form[field_name]) }}
                         {% else %}
-                            <em>{{ 'message_form_group_empty'|trans({}, 'SonataAdminBundle') }}</em>
+                            {% if form_group.empty_message is not defined or form_group.empty_message != false %}
+                                <em>{{ (form_group.empty_message ?? 'message_form_group_empty')|trans({}, 'SonataAdminBundle') }}</em>
```
Comment (core23) [inline src/Resources/views/CRUD/base_edit_form_macro.html.twig]:
```suggestion
                                <em>{{ form_group.empty_message|default('message_form_group_empty')|trans({}, 'SonataAdminBundle') }}</em>
```

**Comment by RobinDev** [comment]
New try.

**Comment by VincentLanglet** [comment]
@RobinDev You just have to update the tests and it's ok for me

**Comment by franmomu** [comment]
Thanks @RobinDev !

**Pull request merged**